### PR TITLE
[Merged by Bors] - feat: port Analysis.LocallyConvex.Basic

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/LocallyConvex/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+theorem balanced_Union₂ {f : ∀ i, κ i → Set E} (h : ∀ i j, Balanced 𝕜 (f i j)) :
+    Balanced 𝕜 (⋃ (i) (j), f i j) :=
+  balanced_unionᵢ fun _ => balanced_unionᵢ <| h _
+#align balanced_Union₂ balanced_Union₂
+
+theorem balanced_interᵢ {f : ι → Set E} (h : ∀ i, Balanced 𝕜 (f i)) : Balanced 𝕜 (⋂ i, f i) :=
+  fun _a ha => (smul_set_interᵢ_subset _ _).trans <| interᵢ_mono fun _ => h _ _ ha
+#align balanced_Inter balanced_interᵢ
+
+theorem balanced_Inter₂ {f : ∀ i, κ i → Set E} (h : ∀ i j, Balanced 𝕜 (f i j)) :
```
Comment (dupuisf) [inline Mathlib/Analysis/LocallyConvex/Basic.lean:203]:
```suggestion
theorem balanced_interᵢ₂ {f : ∀ i, κ i → Set E} (h : ∀ i j, Balanced 𝕜 (f i j)) :
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/LocallyConvex/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+theorem Balanced.inter (hA : Balanced 𝕜 A) (hB : Balanced 𝕜 B) : Balanced 𝕜 (A ∩ B) := fun _a ha =>
+  smul_set_inter_subset.trans <| inter_subset_inter (hA _ ha) <| hB _ ha
+#align balanced.inter Balanced.inter
+
+theorem balanced_unionᵢ {f : ι → Set E} (h : ∀ i, Balanced 𝕜 (f i)) : Balanced 𝕜 (⋃ i, f i) :=
+  fun _a ha => (smul_set_Union _ _).subset.trans <| unionᵢ_mono fun _ => h _ _ ha
+#align balanced_Union balanced_unionᵢ
+
+theorem balanced_Union₂ {f : ∀ i, κ i → Set E} (h : ∀ i j, Balanced 𝕜 (f i j)) :
```
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Analysis/LocallyConvex/Basic.lean:194]: oops, thanks.

**Pull request closed without merging**